test/studies/ml/lib/Chai.chpl

```diff
@@ -137,32 +137,32 @@ module Chai {
 
         proc init() { }
 
-        proc forwardProp(x: Tensor(?rank)): Tensor(rank) {
+        proc ref forwardProp(x: Tensor(?rank)): Tensor(rank) {
             return tn.sigmoid(x);
         }
 
-        proc forwardPropBatch(batch: [] ?tensorType) where isSubtype(tensorType, Tensor) {
-            return tn.sigmoid(batch);
+        proc ref forwardPropBatch(batch: [] ?tensorType) where isSubtype(tensorType, Tensor) {
+            return forwardProp(batch);
         }
 
-        proc backward(delta: Tensor(?rank),lastInput: Tensor(rank)): Tensor(rank) {
+        proc const ref backward(delta: Tensor(?rank),lastInput: Tensor(rank)): Tensor(rank) {
             return delta * tn.sigmoidPrime(lastInput); 
         }
 
-        proc backwardBatch(deltas: [] ?tensorType1, inputs: [] ?tensorType2) 
+        proc const ref backwardBatch(deltas: [] ?tensorType1, inputs: [] ?tensorType2) 
             where isSubtype(tensorType1, Tensor) && isSubtype(tensorType2, Tensor) {
-            return [(d,i) in zip(deltas,inputs)] backward(d,i);
+            return backward(deltas,inputs);
```

Comment:
Same here. So I'm guessing that's intentional

Reply:
Yes, this seemed to alleviate some of the `ref` errors.